src/mds/Locker.cc

```diff
@@ -2991,6 +2992,10 @@ void Locker::_update_cap_fields(CInode *in, int dirty, MClientCaps *m, inode_t *
 	      << " for " << *in << dendl;
       pi->mtime = mtime;
     }
+    if (((dirty & CEPH_CAP_FILE_WR) && change_attr > pi->change_attr) ||
+	((dirty & CEPH_CAP_FILE_EXCL) && change_attr != pi->change_attr)) {
+      pi->change_attr = change_attr;
```

Comment:
Ok, I'll look at that too. FWIW, what ganesha does currently is fake up the change_attr from the ctime when the underlying storage engine doesn't have a real change attr. So we could follow suit here -- bump the change_attr if the client didn't return a change_attr with the caps, and it looks like the ctime has changed.